src/reconcile/rec_child.c

```diff
@@ -125,6 +133,33 @@ __rec_child_deleted(
      * cells to the page. Copy out the current fast-truncate information for that function.
      */
     if (!visible_all) {
+        if (__wt_process.fast_truncate_2022) {
+            /*
+             * We rely on the next reconciliation to remove the deleted entries if they become
+             * globally visible. Mark the page dirty for checkpoint to ensure we will trigger the
+             * next reconciliation on this page. Otherwise, the deleted pages may never be removed
+             * from disk. For eviction, the page is marked dirty the next time it is read into the
+             * memory.
+             */
+            if (!F_ISSET(r, WT_REC_EVICT))
+                r->leave_dirty = true;
+        } else {
+            /*
+             * Internal pages with deletes that aren't globally visible cannot be evicted if we
+             * don't write the page_del information, we don't have sufficient information to restore
+             * the page's information if subsequently read (we wouldn't know which transactions
+             * should see the original page and which should see the deleted page).
+             */
+            if (F_ISSET(r, WT_REC_EVICT))
+                return (__wt_set_return(session, EBUSY));
+
+            /*
+             * It is wrong to leave the page clean in checkpoint if we cannot write the deleted
+             * pages to disk in eviction. If we do so, the next eviction will write the non-globally
```

Comment:
`to disk in eviction` to `to disk in checkpoint`.

How about `the next eviction will lose the time point information of non-obsolete deleted pages`?